proof/proof.go

```diff
@@ -296,6 +297,12 @@ type Proof struct {
 	// a valid signature to spend the asset.
 	ChallengeWitness wire.TxWitness
 
+	// ChallengeBytes is an optional field of the proof that is used when
+	// verifying an ownership proof. This fiels is only populated when the
+	// corresponding ProofVerificationOption option is defined. This is
+	// not encoded in the proof's records.
+	ChallengeBytes fn.Option[[32]byte]
```

Comment:
I think we shouldn't transport this in the `Proof` struct, as that will be confusing since we don't encode it.

Instead we can just have a struct `proofVerificationParams` with the challenge bytes that is being set by the `ProofVerificationOption` function.